cmake/KokkosConfig.cmake.in

```diff
@@ -69,4 +69,26 @@ elseif(@Kokkos_ENABLE_CUDA@ AND NOT @KOKKOS_COMPILE_LANGUAGE@ STREQUAL CUDA AND
   unset(IS_NVCC)
 endif()
 
+if(@Kokkos_ENABLE_COMPILE_AS_CMAKE_LANGUAGE@)
+  if(NOT (${CMAKE_@KOKKOS_COMPILE_LANGUAGE@_ARCHITECTURES} STREQUAL ${Kokkos_@KOKKOS_COMPILE_LANGUAGE@_ARCHITECTURES}))
+    message(
+      WARNING
+        "Kokkos enabled the architecture '${CMAKE_@KOKKOS_COMPILE_LANGUAGE@_ARCHITECTURES}' which is not the same the  is not the same architecture as the architectures enabled in CMake '${CMAKE_@KOKKOS_COMPILE_LANGUAGE@_ARCHITECTURES}'.
+        This
+        can
+        lead
+        to
+        problems
+        when
+        Kokkos
+        and
+        @KOKKOS_COMPILE_LANGUAGE@
+        code
+        is
+        mixed.
+        "
```

Comment:
I hope `cmake-format` will do something better after putting in all in one line first.
```suggestion
        "Kokkos enabled the architecture '${CMAKE_@KOKKOS_COMPILE_LANGUAGE@_ARCHITECTURES}' which is not the same the  is not the same architecture as the architectures enabled in CMake '${CMAKE_@KOKKOS_COMPILE_LANGUAGE@_ARCHITECTURES}'. This can lead to problems when Kokkos and @KOKKOS_COMPILE_LANGUAGE@ code is mixed."
```

Reply:
fixed in https://github.com/kokkos/kokkos/pull/8166/commits/7751e7a7b812271db6c7d3b2797334ac9db0935f ... not sure what glitched there